packages/@uppy/provider-views/src/Browser.js

```diff
@@ -1,57 +1,139 @@
-const classNames = require('classnames')
 const { h } = require('preact')
+const classNames = require('classnames')
+
+const remoteFileObjToLocal = require('@uppy/utils/lib/remoteFileObjToLocal')
+
 const Filter = require('./Filter')
-const ItemList = require('./ItemList')
 const FooterActions = require('./FooterActions')
+const Item = require('./Item/index')
 
-const Browser = (props) => {
+const VIRTUAL_SHARED_DIR = 'shared-with-me'
+
+function Browser (props) {
   const {
     currentSelection,
     folders,
     files,
     uppyFiles,
-    filterItems,
+    viewType,
+    headerComponent,
+    showBreadcrumbs,
+    isChecked,
+    toggleCheckbox,
+    handleScroll,
+    showTitles,
+    i18n,
+    validateRestrictions,
+    showFilter,
+    filterQuery,
     filterInput,
+    getNextFolder,
+    cancel,
+    done,
+    columns,
   } = props
 
-  let filteredFolders = folders
-  let filteredFiles = files
-
-  if (filterInput !== '') {
-    filteredFolders = filterItems(folders)
-    filteredFiles = filterItems(files)
-  }
-
   const selected = currentSelection.length
 
   return (
-    <div className={classNames('uppy-ProviderBrowser', `uppy-ProviderBrowser-viewType--${props.viewType}`)}>
+    <div
+      className={classNames(
+        'uppy-ProviderBrowser',
+        `uppy-ProviderBrowser-viewType--${viewType}`
+      )}
+    >
       <div className="uppy-ProviderBrowser-header">
-        <div className={classNames('uppy-ProviderBrowser-headerBar', !props.showBreadcrumbs && 'uppy-ProviderBrowser-headerBar--simple')}>
-          {props.headerComponent}
+        <div
+          className={classNames(
+            'uppy-ProviderBrowser-headerBar',
+            !showBreadcrumbs && 'uppy-ProviderBrowser-headerBar--simple'
+          )}
+        >
+          {headerComponent}
         </div>
       </div>
-      {props.showFilter && <Filter {...props} />}
-      <ItemList
-        columns={[{
-          name: 'Name',
-          key: 'title',
-        }]}
-        folders={filteredFolders}
-        files={filteredFiles}
-        isChecked={props.isChecked}
-        handleFolderClick={props.getNextFolder}
-        toggleCheckbox={props.toggleCheckbox}
-        handleScroll={props.handleScroll}
-        title={props.title}
-        showTitles={props.showTitles}
-        i18n={props.i18n}
-        viewType={props.viewType}
-        validateRestrictions={props.validateRestrictions}
-        uppyFiles={uppyFiles}
-        currentSelection={currentSelection}
-      />
-      {selected > 0 && <FooterActions selected={selected} {...props} />}
+
+      {showFilter && (
+        <Filter
+          i18n={i18n}
+          filterQuery={filterQuery}
+          filterInput={filterInput}
+        />
+      )}
+
+      {(() => {
+        if (!folders.length && !files.length) {
+          return (
+            <div className="uppy-Provider-empty">
+              {props.i18n('noFilesFound')}
+            </div>
+          )
+        }
+
+        return (
```

Comment:
We should either use a different component, or use a ternary. IIRC using an arrow function makes it impossible for (p)react to re-use the DOM objects because the arrow functions always return a different object reference.
```suggestion
      {(!folders.length && !files.length) ? (
            <div className="uppy-Provider-empty">
              {props.i18n('noFilesFound')}
            </div>
          ) : (
```

Reply:
To the best of my knowledge that is not true and this is a valid pattern. It's not any different from this pattern, which we use all the time:

```js

renderSomething() {
  if (something) {
    return ....
  }
  return ...
}

render() {
  return (
    <div>{renderSomething()}</div>
  )
}
```

I prefer this for readability. It's also used here: 

https://github.com/transloadit/uppy/blob/a298e59f17d6a1aba7786fe8c60df73242104940/packages/%40uppy/status-bar/src/StatusBar.js#L173-L208